### ISSUE TITLE
Introduce GitHub Actions workflow to build Ruby with MMTk

### ISSUE THREAD

**Issue by eightbitraptor** [mmtk/ruby#52]
I propose adding a GitHub actions workflow to this fork to build Ruby with MMTk.

Having this will likely expose complexities in the build process that will potentially need to be addressed before we can upstream a patch.

We may be able to build on the work done by [The YJIT build workflow](https://github.com/ruby/ruby/blob/master/.github/workflows/yjit-ubuntu.yml) to help us achieve this.

**Comment by wks** [comment]
Good idea. It will also enable us to do binding tests on Ruby when mmtk-core introduces an API change.
